insights/core/spec_factory.py

```diff
@@ -452,14 +455,18 @@ def simple_command(cmd, context=HostContext, split=True, keep_rc=False, timeout=
     def inner(broker):
         ctx = broker[context]
         rc = None
-        raw = ctx.shell_out(cmd, split=split, keep_rc=keep_rc, timeout=timeout)
+        if split:
+            filters = "|".join(get_filters(inner))
+        if filters:
+            raw = ctx.shell_out("{0} | grep -E '{1}'".format(cmd, filters), split=split, keep_rc=keep_rc,
```

Comment:
Have you considered chaining `Popen` objects together rather than relying on a shell to do the pipelining?

Reply:
Thanks Jesse,
I actually do that further down in the calls. shell_out calls check_output in the ExecutionContext. Check_output then calls the call function in supbroc and the subproc class does not support chaining Popens. To get around that I override check_output in the HostContext and it parses the arguments and does the chaining of the Popens (does not uses subproc).
I was trying not to change current behavior in the upper functions that other classes might be using.
Hope all this babble makes sense.

Reply:
> I was trying not to change current behavior in the upper functions that other classes might be using.

I think making your changes in subproc makes the most sense to me.  Having code duplication like this is probably a maintenance burden long-term.  We can always perform regression testing to ensure you don't break backwards compat.  As long as you aren't changing/breaking APIs used by plugin repos.

```
$ ag 'subproc[^e]' -l insights
insights/core/context.py
insights/core/spec_factory.py
insights/core/archives.py
insights/tests/test_subproc.py
insights/util/fs.py
```

Not too many consumers of `subproc` it seems.

Reply:
That sounds good to me, just trying not to stir up too much dust. Removing the override if check_output) in the context class will eliminate the the issue you mentioned about...
STDOUT = subprocess.PIPE
STDERR = subprocess.STDOUT
While I'm at it I'll see if I can change that in subproc().